crowbar_framework/app/controllers/installer_controller.rb

```diff
@@ -0,0 +1,135 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class InstallerController < ApplicationController
+  skip_before_filter :crowbar_mode
+  before_filter :hide_navigation
+
+  def index
+    @steps = all_steps
+  end
+
+  #
+  # Perform Crowbar Installation
+  #
+  # Provides the restful api call for
+  # /installer/status 	GET 	return done steps and error
+  # returns a hash with an indicator if the installation failed and the steps that are done
+  def status
+    if successful?
+      ENV["CROWBAR_MODE"] = nil if ENV["CROWBAR_MODE"]
+      respond_to do |format|
+        format.json { redirect_to root_path }
+        format.html { redirect_to root_path }
+      end
+    else
+      respond_to do |format|
+        format.json { render json: steps_done }
+        format.html { redirect_to installer_url }
+      end
+    end
+  end
+
+  #
+  # Set the status of an installation step
+  #
+  # Provides the restful api call for
+  # /installer/set_status 	POST 	set the status
```

Comment:
Why `/installer/set_status POST`? It's different than `/installer/status GET`, so just do `/installer/status POST` instead of `set_status`.